packages/core/components/AnnotationFilterForm/index.tsx

```diff
@@ -85,10 +132,55 @@ export default function AnnotationFilterForm(props: AnnotationFilterFormProps) {
                         : props.annotation.valueOf(item.value)
                 )
         );
+        removeSpecialFilters();
         dispatch(selection.actions.addFileFilter(filters));
     };
 
+    // Any value/"include"
+    const onSelectAny = () => {
```

Comment:
Idk if it's worth it since its just two options but there's a lot of reused stuff here. could combine to something like 
```
function applyFilter(filterType: "include" | "exclude") {
    removeSpecialFilters();
    dispatch(selection.actions.removeFuzzyFilter(new FuzzyFilter(props.annotation.name)));
    onDeselectAll();

    if (filterType === "include" && !usingAnyValueFilter) {
        const includeFilter = new IncludeFilter(props.annotation.name);
        dispatch(selection.actions.addIncludeFilter(includeFilter));
    } else if (filterType === "exclude" && !usingNoValueFilter) {
        const excludeFilter = new ExcludeFilter(props.annotation.name);
        dispatch(selection.actions.addExcludeFilter(excludeFilter));
    }
}

const onSelectAny = () => applyFilter("include");
const onSelectNone = () => applyFilter("exclude");
```
Or alternatively moving the dispatch and onDeselect all into removeSpecialFilters() Also afterthought, would there be possibility for more filter types in the future?

Reply:
Only minor concern with this solution is we'd be doing extra dispatches if the filter is already applied, which is why I originally had them in the conditionals. Not going to break anything, but would be extra unnecessary calls to state.
Similarly, would prefer not to move onDeselectAll into removeSpecialFilters with how we currently have it, since it's also called in situations where we don't do a deselectAll, so we'd also end up doing extra dispatches.
This whole file can definitely use a substantive refactor though, the inter-function dependencies are pretty messy

Reply:
Slightly refactored this into one function in [69f2d81](https://github.com/AllenInstitute/biofile-finder/pull/238/commits/69f2d81cd0bc08021ab4bc01108dcd64a4a5e048). Not 100% sure how much of improvement it is